alma/tests/Unit/lib/Utils/SettingsCustomFieldsTest.php

```diff
@@ -0,0 +1,92 @@
+<?php
+namespace Tests\Unit\lib\Utils;
+
+use Mockery;
+use Language;
+use PHPUnit\Framework\TestCase;
+use Alma\PrestaShop\Utils\SettingsCustomFields;
+use Mockery\Adapter\Phpunit\MockeryPHPUnitIntegration;
+
+define('_PS_MODULE_DIR_', '../');
+define('_PS_THEME_DIR_', '../');
+
+class SettingsCustomFieldsTest extends TestCase
+{
+    use MockeryPHPUnitIntegration;
+    // public function testExemple()
+    // {
+    //     $array = [];
+
+    //     $expected = 0;
+
+    //     $this->assertSame($expected, count($array));
+    // }
+
+    // public function tearDown()
+    // {
+    //     Mockery::close();
+    // }
+
+    public function testGetCustomFieldButtonTitle2LangsFrEn()
+    {
+        $arrayLanguages = [
+            [
+                'id_lang' => '1',
+                'name' => 'Français (French)',
+                'active' => '1',
+                'iso_code' => 'fr',
+                'language_code' => 'fr',
+                'locale' => 'fr-FR',
+                'date_format_lite' => 'd/m/Y',
+                'date_format_full' => 'd/m/Y H:i:s',
+                'is_rtl' => '0',
+                'id_shop' => '1',
+                'shops' => [
+                    1 => true
+                ],
+            ],
+            [
+                'id_lang' => '2',
+                'name' => 'English (English)',
+                'active' => '1',
+                'iso_code' => 'en',
+                'language_code' => 'en-us',
+                'locale' => 'en-US',
+                'date_format_lite' => 'd/m/Y',
+                'date_format_full' => 'd/m/Y H:i:s',
+                'is_rtl' => '0',
+                'id_shop' => '1',
+                'shops' => [
+                    1 => true
+                ],
+            ]
+        ];
+        $mock = Mockery::mock('Language');
+        $mock->shouldReceive('language')
+            ->andReturn($arrayLanguages);
+        $languages = $mock->language();
```

Comment:
you should not call `language()` yourself if it have to be called inside the method you are testing

the method you want to test here (and should be tested in another test method) seem's to be :

`Language::getLanguages(false)` from `initCustomFields()` (or maybe `aggregateAllLanguagesCustomFields($keyConfig)`) ... no ?

Reply:
I probably didn't understand everything about the mock, some explanations in pair would be nice